rasa_sdk/endpoint.py

```diff
@@ -39,120 +54,63 @@ def create_argument_parser():
-            response.status_code = 400
-            return response
-
-        return jsonify(response)
-
-    @app.route("/actions", methods=["GET", "OPTIONS"])
-    @cross_origin(origins=cors_origins)
-    def actions():
+            logger.error(e)
+            body = {"error": e.message, "action_name": e.action_name}
+            return response.json(body, status=400)
+        except ActionNotFoundException as e:
+            logger.error(e)
+            body = {"error": e.message, "action_name": e.action_name}
+            return response.json(body, status=404)
+
+        return response.json(result, status=200)
+
+    @app.get("/actions")
+    async def actions(request):
         """List all registered actions."""
-        return jsonify([{"name": k} for k in executor.actions.keys()])
+        body = [{"name": k} for k in executor.actions.keys()]
+        return response.json(body, status=200)
 
     return app
 
 
-def check_version_compatibility(rasa_version):
-    """Check if the version of rasa and rasa_sdk are compatible.
-
-    The version check relies on the version string being formatted as
-    'x.y.z' and compares whether the numbers x and y are the same for both
-    rasa and rasa_sdk.
-    Args:
-        rasa_version - A string containing the version of rasa that
-        is making the call to the action server.
-    Raises:
-        Warning - The version of rasa version unknown or not compatible with
-        this version of rasa_sdk.
-    """
-    # Check for versions of Rasa that are too old to report their version number
-    if rasa_version is None:
-        logger.warning(
-            "You are using an old version of rasa which might "
-            "not be compatible with this version of rasa_sdk "
-            "({}).\n"
-            "To ensure compatibility use the same version "
-            "for both, modulo the last number, i.e. using version "
-            "A.B.x the numbers A and B should be identical for "
-            "both rasa and rasa_sdk."
-            "".format(rasa_sdk.__version__)
-        )
-        return
-
-    rasa = rasa_version.split(".")[:-1]
-    sdk = rasa_sdk.__version__.split(".")[:-1]
-
-    if rasa != sdk:
-        logger.warning(
-            "Your versions of rasa and "
-            "rasa_sdk might not be compatible. You "
-            "are currently running rasa version {} "
-            "and rasa_sdk version {}.\n"
-            "To ensure compatibility use the same "
-            "version for both, modulo the last number, "
-            "i.e. using version A.B.x the numbers A and "
-            "B should be identical for "
-            "both rasa and rasa_sdk."
-            "".format(rasa_version, rasa_sdk.__version__)
-        )
-
-
 def run(
-    action_package_name,
-    port=DEFAULT_SERVER_PORT,
-    cors_origins="*",
-    ssl_certificate=None,
-    ssl_keyfile=None,
-    ssl_password=None,
+    action_package_name: Union[Text, types.ModuleType],
+    port: int = DEFAULT_SERVER_PORT,
```

Comment:
I think this can also be a text (doesn't needs to be handled, I tried)
```suggestion
    port: Union[Text, int] = DEFAULT_SERVER_PORT,
```